pkg/server/plugin/upstreamauthority/gcpca/gcpca.go

```diff
@@ -0,0 +1,284 @@
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	pcaapi "cloud.google.com/go/security/privateca/apiv1beta1"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/common/x509util"
+	"github.com/spiffe/spire/pkg/server/plugin/upstreamauthority"
+	"github.com/spiffe/spire/proto/spire/common/plugin"
+	"github.com/zeebo/errs"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+	privatecapb "google.golang.org/genproto/googleapis/cloud/security/privateca/v1beta1"
+)
+
+const (
+	// The name of the plugin
+	pluginName = "gcp_ca"
+	// The header and footer type for a PEM-encoded CSR
+	csrRequestType = "CERTIFICATE REQUEST"
+)
+
+// KeyUsage represents the set of actions that are valid for a given key. It's
+// a bitmap of the KeyUsage* constants.
+type KeyUsage int
+
+const (
+	KeyUsageDigitalSignature KeyUsage = 1 << iota
+	KeyUsageContentCommitment
+	KeyUsageKeyEncipherment
+	KeyUsageDataEncipherment
+	KeyUsageKeyAgreement
+	KeyUsageCertSign
+	KeyUsageCRLSign
+	KeyUsageEncipherOnly
+	KeyUsageDecipherOnly
+)
+
+var (
+	// pluginErr is a convenience error class that prefixes errors with the
+	// plugin name.
+	pluginErr = errs.Class(pluginName)
+)
+
+// BuiltIn constructs a catalog Plugin using a new instance of this plugin.
+func BuiltIn() catalog.Plugin {
+	return builtin(New())
+}
+
+func builtin(p *Plugin) catalog.Plugin {
+	return catalog.MakePlugin(pluginName, upstreamauthority.PluginServer(p))
+}
+
+type Config struct {
+	Project  string `hcl:"gcp_project_name"`
+	Location string `hcl:"gcp_region_name"`
+	CertName string `hcl:"gcp_cert_name"`
+}
+
+type Plugin struct {
+	// gRPC requires embedding either the "Unimplemented" or "Unsafe" stub as
+	// a way of opting in or out of forward build compatibility.
+	upstreamauthority.UnimplementedUpstreamAuthorityServer
+
+	// mu is a mutex that protects the configuration. Plugins may at some point
+	// need to support hot-reloading of configuration (by receiving another
+	// call to Configure). So we need to prevent the configuration from
+	// being used concurrently and make sure it is updated atomically.
+	mu sync.Mutex
+	c  *Config
+
+	log hclog.Logger
+}
+
+// These are compile time assertions that the plugin matches the interfaces the
+// catalog requires to provide the plugin with a logger and host service
+// broker as well as the UpstreamAuthority itself.
+var _ catalog.NeedsLogger = (*Plugin)(nil)
+var _ catalog.NeedsLogger = (*Plugin)(nil)
+var _ upstreamauthority.UpstreamAuthorityServer = (*Plugin)(nil)
+
+func New() *Plugin {
+	return &Plugin{}
+}
+
+// SetLogger will be called by the catalog system to provide the plugin with
+// a logger when it is loaded. The logger is wired up to the SPIRE core
+// logger
+func (p *Plugin) SetLogger(log hclog.Logger) {
+	// TODO: store the logger for later use. If the plugin does not need to
```

Comment:
This comment from the "example" template can be removed, since you are certainly using the logger.